lib/daru/dataframe.rb

```diff
@@ -263,6 +264,74 @@ def [](*names)
       dispatch_to_axis axis, :access, *names
     end
 
+    # Retrive rows by positions
+    # @param [Array<Integer>] *positions positions of rows to retrive
+    # @return rows specified by positions
+    def at *positions
+      if positions.size == 1
+        return Daru::Vector.new @data.map { |vec| vec.at(*positions) },
+          index: @vectors
+      else
+        new_rows = @data.map { |vec| vec.at(*positions) }
+        return Daru::DataFrame.new new_rows,
+          index: @index.at(*positions),
+          order: @vectors
+      end
+    end
+
+    # Set rows by positions
+    # @param [Array<Integer>] positions positions of rows to set
+    # @vector [Array, Daru::Vector] vector vector to be assigned
+    def row_at_set positions, vector
+      vector =
+        if vector.is_a? Daru::Vector
+          vector.reindex @vectors
+        else
+          Daru::Vector.new vector
+        end
+
+      raise SizeError, 'Vector length should match row length' if
+        vector.size != @vectors.size
+
```

Comment:
You can modify `DataFrame#at` in the same way that `DataFrame#[]` works: pass `:row` as a second argument.